packages/toolpad-app/src/exports/next.tsx

```diff
@@ -0,0 +1,18 @@
+'use server';
+
+import * as React from 'react';
+import ToolpadAppClient from './next-client';
+import { loadDom } from '../server/localMode';
+import createRuntimeState from '../runtime/createRuntimeState';
+
+export interface ToolpadAppServerProps {
+  base: string;
+  dir: string;
+}
+
+export async function ToolpadApp({ base, dir = './toolpad' }: ToolpadAppServerProps) {
```

Comment:
How does the app decide which page to run? If it's the URL path concerning `base`, is it possible to abstract the URL from the app and modify the props as follows?

```
export interface ToolpadAppServerProps {
  dir: string;
  page: string;
}
```

It could help us dynamically render any page without worrying about the paths, giving user more control IMO. WDYT?

Reply:
Yes, that makes sense. This is somewhat on hold because it depends on a bunch of [low-level work](https://github.com/mui/mui-toolpad/pull/3171) that needs to happen first. We'll also have to abstract Toolpad routing to be able to plug in the Next.js router.

Reply:
> Another option I'm thinking of is to expose another component that renders a single page instead of the whole app.
Yes, that would be even more useful for [my use case](https://github.com/mui/mui-toolpad/issues/315#issuecomment-1805702316) where I have a backend to pull the MDX file and I can transform it on the fly render the page without even shell. Further, the users can use Toolpad to generate the yml where I can provide a serializer for them.